### PR TITLE
Add getApkUrlAsync and cancelBuild

### PR DESCRIPTION
**Pull request by cpirich**
* This PR includes all of the changes in this previous PR that will be closed once this is approved and merged: https://github.com/expo/snack-sdk/pull/9
* It adds some improved error handling and a new `cancelBuild` API that is implemented using `graphql-request` in this commit: https://github.com/expo/snack-sdk/commit/989e11bcc028dd3bd97081d9648ac868bd743fed

### CODE REVIEW

File: src/SnackSession.js [hunk 1]
```diff
@@ -292,6 +299,148 @@ export default class SnackSession {
...
+      }
+    } catch (e) {
+      console.error(e);
+      throw e;
+    }
+  }
+
+  cancelBuild = async (buildId) => {
+
+    const url = 'https://expo.io/--/graphql';
```
Comment (tcdavis) [inline src/SnackSession.js]: https://expo.io should be this.expoApiUrl

File: src/SnackSession.js [hunk 2]
```diff
@@ -292,6 +299,148 @@ export default class SnackSession {
     return url;
   };
 
+  /**
+   * Generates a default app.json template for the snack.
+   * @returns {Object} the app.json template
+   * @function
+   */
+  generateAppJson = (): Object => {
```
Comment (tcdavis) [inline src/SnackSession.js]: I'd remove this method

Comment (cpirich) [reply src/SnackSession.js]: ok with me, as we don't use it

**Comment by cpirich** [comment]
Closing as these changes are now in a separate library (https://github.com/expo/snack-build)

**Pull request closed without merging**